client/lib/create-cached-selector/README.md

```diff
@@ -0,0 +1,70 @@
+`create-cached-selector`
+=================
+
+This module exports a function which creates a cached state selector for use with the Redux global application state. It is a good idea to use this function over plain selectors whenever either the computation over state or React's rerenders are expensive.
+
+## Usage
+
+`createCachedSelector` accepts the following arguments:
+
+- **getDependents**: A function which maps over `state` and returns all of the relevant parts of state the selector needs. You should be creating an object whose values are all the return of other selectors -- no computations allowed here. 
+- **selector**: A function which takes in the same args as `getDependents` with one catch. Instead of being passed state as its first arg, it is given the results of getDependents.  This forces you to declare all of your state-dependencies.
+
+For example, imagine that our state contains post objects, each of which are assigned to a particular site. Retrieving an array of posts for a specific site would require us to filter over all of the known posts. While this would normally trigger a re-render in React, we can use `createCachedSelector` to create a cached version of the selector which can return a referentially equal object:
+
+```js
+const getDependents = state => ( { posts: state.posts } );
+const selector = ( { posts }, siteId ) => filter( posts, { siteId } );
+
+const getSitePosts = createCachedSelector( selector, getDependents );
```

Comment:
wouldn't it make more sense to pass `getDependents` first since they are logically prior to the selector?

this isn't a big deal either way, but something feels odd about this ordering

Reply:
I agree with you wholeheartedly.  I only picked this order to match createSelector's api.
Since we are diverging anyway, may as well diverge a bit more ;).

Updated!